File: src/main/java/com/icloud/command/before/Button.java
package com.icloud.command.before;

public class Button {

    private final Light light;

    public Button(Light light) {
        this.light = light;
    }

    public void press() {
        light.on();
    }

    public static void main(String[] args) {
        Button button = new Button(new Light());
        button.press();
        button.press();
        button.press();
        button.press();
    }
}

File: src/main/java/com/icloud/command/review/Button.kt
package com.icloud.command.review

import java.util.Stack

class Button {

    private var commands: Stack<Command> = Stack()

    fun press(command: Command) {
        command.execute()
        commands.push(command)
    }

    fun undo() {
        if (!commands.isEmpty()) {
            commands.pop().undo()
        }
    }

}

fun main() {

    val button: Button = Button()
    button.press(GameStartCommand(Game()))
    button.press(LightOnCommand(Light()))
    button.undo()
    button.undo()
}